addons/im_livechat/static/src/core/common/@types/models.d.ts

```diff
@@ -1,18 +1,50 @@
-import { ChatbotScriptStep as ChatbotScriptStepClass } from "@im_livechat/core/common/chatbot_script_step_model";
-import { ChatbotStep as ChatbotStepClass } from "@im_livechat/core/common/chatbot_step_model";
-import { Chatbot as ChatbotClass } from "@im_livechat/core/common/chatbot_model";
-import { ChatbotScriptStepAnswer as ChatbotScriptStepAnswerClass } from "@im_livechat/core/common/chatbot_script_step_answer_model";
-import { ChatbotScript as ChatbotScriptClass } from "@im_livechat/core/common/chatbot_script_model";
-import { LivechatChannelRule as LivechatChannelRuleClass } from "@im_livechat/core/common/livechat_channel_rule_model";
-
 declare module "models" {
+    import { Chatbot as ChatbotClass } from "@im_livechat/core/common/chatbot_model";
+    import { ChatbotScript as ChatbotScriptClass } from "@im_livechat/core/common/chatbot_script_model";
+    import { ChatbotScriptStep as ChatbotScriptStepClass } from "@im_livechat/core/common/chatbot_script_step_model";
+    import { ChatbotScriptStepAnswer as ChatbotScriptStepAnswerClass } from "@im_livechat/core/common/chatbot_script_step_answer_model";
+    import { ChatbotStep as ChatbotStepClass } from "@im_livechat/core/common/chatbot_step_model";
+    import { LivechatChannel as LivechatChannelClass } from "@im_livechat/core/common/livechat_channel_model";
+    import { LivechatChannelRule as LivechatChannelRuleClass } from "@im_livechat/core/common/livechat_channel_rule_model";
+
+    export interface Chatbot extends ChatbotClass {}
+    export interface ChatbotScript extends ChatbotScriptClass {}
+    export interface ChatbotScriptStep extends ChatbotScriptStepClass {}
+    export interface ChatbotScriptStepAnswer extends ChatbotScriptStepAnswerClass {}
+    export interface ChatbotStep extends ChatbotStepClass {}
+    export interface LivechatChannel extends LivechatChannelClass {}
+    export interface LivechatChannelRule extends LivechatChannelRuleClass {}
+
+    export interface ChatWindow {
+        livechatStep: undefined|"CONFIRM_CLOSE"|"FEEDBACK";
+    }
+    export interface Message {
+        chatbotStep: ChatbotStep;
+    }
+    export interface Store {
+        Chatbot: typeof Chatbot;
```

Comment:
It needs to be `typeof ChatbotClass` (same for all of them).

Reply:
Now that I think about it, we can remove insert/get/records overrides in favor of types. I updated the PR.